src/hooks/UseComponentToggleVisible.tsx

```diff
@@ -0,0 +1,33 @@
+import { RefObject, useEffect, useRef, useState } from "react";
+
+interface UseComponentVisibleOutput {
+  ref: RefObject<HTMLDivElement | null>;
+  isComponentVisible: boolean;
+  setIsComponentVisible: Function;
+}
+
+const UseComponentToggleVisible = (
```

Comment:
Feel free to ignore this as out of scope of your PR, but it looks like we could reuse this in `src/design-system/ModalDialog.tsx`, which uses the same logic to close a modal when you click outside of it.

Reply:
yea I saw that as well. Since the modal does a full-screen opacity, the click listener is just on that div rather than attached to the document like in my hook.

I agree it'd be nice to re-use this new hook on the modal itself. Would you prefer I work on refactoring that? or move on to other VQA from the #167 list?

Reply:
Actually we have another feature we want you to move onto after this is merged. We will discuss it in standup.